### PR TITLE
[Merge into #17] [Feature] Purchase Tracking and ViewDetail Logging

### PR DESCRIPTION
**Pull request by yoyoyoseob**
Adds protocols for purchase tracking and detail viewing functionality

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,8 +87,10 @@ First, all analytics trackers implement this base protocol: `AnalyticsTracking`.
 
 While simply implementing this protocol is all you need for your object to be a valid analytics tracker, you won't be able to do too much with it. This is because Simcoe utilizes an array of various different protocols, each one giving defining the functionality needed to handle that API event. These protocols are:
 
+* `ViewDetailLogging`
```
Comment (MichaelCampbell) [inline README.md]: I started alphabetizing these, please continue

File: README.md [hunk 2]
```diff
@@ -100,8 +102,10 @@ So for your analytics provider to be able to handle page views in the Simcoe fra
 
 Each provider type defined above maps directly to a Simcoe function that will call that method. These are, respectively:
 
+* `logViewDetail`
```
Comment (MichaelCampbell) [inline README.md]: match the order from above

File: README.md (hunk 1 above)
Comment (yoyoyoseob) [reply README.md]: Updated

File: README.md (hunk 2 above)
Comment (yoyoyoseob) [reply README.md]: Updated

File: Simcoe/Simcoe.swift [hunk 3]
```diff
@@ -326,6 +326,39 @@ public final class Simcoe {
         }
```
Comment (MichaelCampbell) [inline Simcoe/Simcoe.swift:326]: I think you may have quoted too much above in [line 271](https://github.com/prolificinteractive/simcoe/pull/19/files#diff-4390d00a82dcc4b10a4d8a3132b44fd0R271), all the text is blue

Comment (yoyoyoseob) [reply Simcoe/Simcoe.swift:326]: Everything looks good in the file itself and it compiles correctly

**Review by MichaelCampbell — APPROVED** [review]

**Pull request merged**